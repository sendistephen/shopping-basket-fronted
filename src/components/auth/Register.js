import React from "react";

const Register = () => {
	return (
		<section class="main center">
			<div class="container">
				<div class="main-content-wrapper">
					<div class="main-content-holder">
						<h2>
							Shopping made <span>easy</span>
						</h2>
						<p>
							Manage your shopping with ease. Create, read, update and delete your shopping basket.
						</p>
					</div>
					<div class="card">
						<p>Donâ€™t have an account? Register here to get started.</p>
						<form>
							<input class="form-control" type="text" placeholder="Name" />
							<input class="form-control" type="email" placeholder="Email" />
							<input class="form-control" type="password" placeholder="Password" />
							<input class="btn btn-register" type="submit" value="Register" />
						</form>
					</div>
				</div>
			</div>
		</section>
	);
};

export default Register;
