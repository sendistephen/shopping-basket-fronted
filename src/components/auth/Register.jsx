import React, { Component } from 'react';
import * as userService from '../../services/userService';
import { toast } from 'react-toastify';

class Register extends Component {
  state = {
    data: {
      name: '',
      email: '',
      password: ''
    },
    errors: {}
  };

  onChange = e => {
    const { name, value } = e.target;
    // clone the state obj
    let data = { ...this.state.data };
    data[name] = value;
    this.setState({ data });
  };
  onSubmit = async e => {
    e.preventDefault();

    try {
      const result = await userService.register(this.state.data);
      if (result) {
        toast.success('User registered successfully');

        this.props.history.push('/login');
      }
    } catch (ex) {
      if (ex.response && ex.response.status === 400) {
        let errors = { ...this.state.errors };
        errors.data = ex.response.data;
        this.setState({ errors });
      }
    }
  };
  render() {
    const { name, email, password } = this.state.data;
    return (
      <div className='main'>
        <div className='d-flex'>
          <a
            className='main-login-btn btn btn-outline-secondary shadow'
            href='#'
          >
            Log in
          </a>
          <div className='main-left-side d-flex'>
            <h1
              className='pl-5 align-items-center align-self-center'
              style={{ fontSize: '68px', fontWeight: '900' }}
            >
              Shopping <br /> Basket.
            </h1>
          </div>
          <div className='main-right-side'>
            <div
              className='card rounded-lg shadow'
              style={{ width: '28rem', borderColor: '#C4CFDB' }}
            >
              <div className='card-body'>
                <p
                  className='lead py-3'
                  style={{
                    color: '#3D4852',
                    fontSize: '18px',
                    fontWeight: 'bold'
                  }}
                >
                  {' '}
                  Register to get started.
                </p>
                <form onSubmit={this.onSubmit}>
                  <div className='form-group'>
                    <input
                      type='text'
                      className='form-control form-rounded'
                      name='name'
                      placeholder='Username'
                      value={name}
                      onChange={this.onChange}
                    />
                  </div>
                  <div className='form-group form-rounded'>
                    <input
                      type='email'
                      className='form-control form-rounded'
                      name='email'
                      id='email'
                      placeholder='Email'
                      value={email}
                      onChange={this.onChange}
                    />
                  </div>
                  <div className='form-group'>
                    <input
                      type='password'
                      className='form-control form-rounded'
                      name='password'
                      id='password'
                      placeholder='Password'
                      value={password}
                      onChange={this.onChange}
                    />
                  </div>
                  <button type='submit' className='btn btn-register btn-block'>
                    Register
                  </button>
                  <hr className='my-5' />
                  <p className='lead text-center' style={{ color: '#ABBBC8' }}>
                    Already have an account?{' '}
                    <a style={{ color: '#0071E2', fontWeight: '400' }} href='#'>
                      Login
                    </a>
                  </p>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Register;
