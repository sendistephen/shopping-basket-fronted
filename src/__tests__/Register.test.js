import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Register from '../components/auth/Register';

describe('Register component', () => {
  const wrapper = shallow(<Register />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has a form', () => {
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
  });
});
