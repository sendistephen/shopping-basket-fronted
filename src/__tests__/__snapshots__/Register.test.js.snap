// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders properly 1`] = `
<div
  className="main"
>
  <div
    className="d-flex"
  >
    <Link
      className="main-login-btn btn btn-outline-secondary shadow"
      to="/login"
    >
      Log in
    </Link>
    <div
      className="main-left-side d-flex"
    >
      <h1
        className="pl-5 align-items-center align-self-center"
        style={
          Object {
            "fontSize": "68px",
            "fontWeight": "900",
          }
        }
      >
        Shopping 
        <br />
         Basket.
      </h1>
    </div>
    <div
      className="main-right-side"
    >
      <div
        className="card rounded-lg shadow"
        style={
          Object {
            "borderColor": "#C4CFDB",
            "width": "28rem",
          }
        }
      >
        <div
          className="card-body"
        >
          <p
            className="lead py-3"
            style={
              Object {
                "color": "#3D4852",
                "fontSize": "18px",
                "fontWeight": "bold",
              }
            }
          >
             
            Register to get started.
          </p>
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control form-rounded"
                name="name"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group form-rounded"
            >
              <input
                className="form-control form-rounded"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control form-rounded"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <button
              className="btn btn-register btn-block"
              type="submit"
            >
              Register
            </button>
            <hr
              className="my-5"
            />
            <p
              className="lead text-center"
              style={
                Object {
                  "color": "#ABBBC8",
                }
              }
            >
              Already have an account?
               
              <Link
                style={
                  Object {
                    "color": "#0071E2",
                    "fontWeight": "400",
                  }
                }
                to="/login"
              >
                Login
              </Link>
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
